content/testing.md

```diff
@@ -0,0 +1,683 @@
+
+ - If you want to write a test that can be run against any running version of your app and verifies at the browser level that the right things happen when you push the right buttons, then you are writing an **acceptance test** (sometimes called "end to end test". Such tests typically try to hook into the application as little as possible, beyond perhaps setting up the right data to run a test against.
+
+ - Finally you may wish to test that your application works under typical load or see how much load it can handle before it falls over. This is called a **load test** or **stress test**. Such tests can be challenging to set up and typically aren't run often but are very important for confidence before a big production launch.
+
+<h3 id="challenges-with-meteor">Challenges of testing in Meteor</h3>
+
+In some ways, testing a Meteor is no different to testing any other client-server, JavaScript heavy application. However, especially compared to a more backend focussed, traditional framework, the client-server divide and the reactive nature of code can add extra challenges to testing.
+
+As Meteor's data system makes it simple to bridge the client-server gap and often allows you to create your application without thinking about how data moves around, it becomes critical to test that your code does actually work correctly across that gap. In traditional frameworks where you spend a lot of time thinking about interfaces between client and server, you can often get away with testing both sides of the interface in isolation.
+
+The good news is that you can easily use Meteor's [full app test mode](#test-modes) to write [integration tests](#full-app-integration-test) that bridge both sides of the gap relatively easily.
+
+Another challenge is creating test data in the client context; we'll discuss ways to do this in the [section on generating test data](#generating-test-data) below. 
+
+As Meteor's reactivity system is "eventually consistent" in the sense that when you change an reactive input to the system, some time later you'll see the user interface change to reflect this. This can be a challenge when testing, however there are some ways to wait until those changes should have happened and verify the results, such as using `Tracker.flush()`.
+
+XXX: say more about this?
+
+<h2 id="test-modes">Test modes in Meteor</h2>
+
+The primary way to test your application in Meteor is the `meteor test` command. 
+
+This loads your application in a special "test mode". What this does is:
+
+ 1. *Doesn't* eagerly load *any* of our application code as Meteor normally would.
```

Comment:
For some reason this list doesn't seem as clear as it could be. Maybe I'll try editing it a bit. But it's critical that people understand exactly what is going on here.